cdap-ui/app/cdap/components/AbstractWidget/SqlSelectorWidget/index.tsx

```diff
@@ -0,0 +1,332 @@
+        }),
+      },
+      () => {
+        // If all of the expansion panels are individually opened or closed,
+        // this will change button expand all button text.
+        let expandedCount = 0;
+        this.state.parsedInputSchemas.forEach((stage) => {
+          if (stage.expanded) {
+            expandedCount++;
+          }
+        });
+        if (expandedCount === this.state.parsedInputSchemas.length) {
+          this.setState({ expandAll: true });
+        } else if (expandedCount === 0) {
+          this.setState({ expandAll: false });
+        }
+      }
+    );
+  };
+
+  public expandAllStages = () => {
+    this.setState({
+      parsedInputSchemas: this.state.parsedInputSchemas.map((stage) => {
+        stage.expanded = true;
+        return stage;
+      }),
+    });
+  };
+
+  public collapseAllStages = () => {
+    this.setState({
+      parsedInputSchemas: this.state.parsedInputSchemas.map((stage) => {
+        stage.expanded = false;
+        return stage;
+      }),
+    });
+  };
+
+  public getStageError = () => {
+    const stageCount = {};
+    this.state.parsedInputSchemas.forEach((input) => {
+      input.schema.forEach((field) => {
+        if (this.state.aliases[field.alias] > 1) {
+          if (!stageCount[input.name]) {
+            stageCount[input.name] = 1;
+          } else {
+            stageCount[input.name]++;
+          }
+        }
+      });
+    });
+    this.setState({
+      errors: {
+        ...this.state.errors,
+        stageCount,
+      },
+    });
+  };
+
+  public formatOutput = () => {
+    const outputArr = [];
+    const tempAliases = {};
+    let errorExists = false;
+    this.state.parsedInputSchemas.forEach((input) => {
+      input.schema.forEach((field) => {
+        if (!field.selected) {
+          return;
+        }
+        const outputField = `${input.name}.${field.name}${field.alias ? ` as ${field.alias}` : ''}`;
+        if (!tempAliases[field.alias]) {
+          tempAliases[field.alias] = 1;
+        } else {
+          tempAliases[field.alias]++;
+          errorExists = true;
+        }
+        outputArr.push(outputField);
+      });
+    });
+    this.setState(
+      {
+        errors: {
+          stageCount: this.state.errors.stageCount,
+          message: this.state.errors.message,
+          exist: errorExists,
+        },
+        aliases: tempAliases,
+      },
+      () => {
+        this.getStageError();
+      }
+    );
+    this.props.onChange(outputArr.join(','));
+  };
+
+  public init = (inputModel) => {
+    const initialModel = {};
+    const parsedInputSchemas: IParsedInputSchema[] = [];
+    if (inputModel) {
+      inputModel.split(',').forEach((entry) => {
+        const split = entry.split(' as ');
+        const fieldInfo = split[0].split('.');
+        if (!initialModel[fieldInfo[0]]) {
+          initialModel[fieldInfo[0]] = {};
+        }
+        initialModel[fieldInfo[0]][fieldInfo[1]] = split[1] ? split[1] : true;
```

Comment:
Might also be better if we check if `fieldInfo` is an array with more than one value. Otherwise , say `fieldInfo[1]` can resolve to `undefined` right?